lib/scala/interpreter-dsl/src/main/java/org/enso/interpreter/dsl/Builtin.java

```diff
@@ -42,10 +44,29 @@
     Class<? extends Exception> from();
     /** @return Class of Enso's builtin (error) type */
     Class<?> to();
+
+    /**
+     * @return true, if only the original exception should be wrapped. Otherwise we pass provided
+     *     method params.
+     */
+    boolean propagate() default false;
   }
 
   /** Container for {@link WrapException} annotations */
   @interface WrapExceptions {
     WrapException[] value() default {};
   }
+
+  /**
+   * Annotation accepting `env.asGUestValue` translation done on the return object. The conversion
+   * is generated automatically. The annotation only ensures that it is intentianal.
+   */
+  @interface ReturningGuestObject {}
```

Comment:
The guest object part of Truffle is going to be deprecated and the point of your work is to stop using it. What do we need this annotation for? Is there a type we cannot replace with your new system?

Reply:
Typo: `asGUestValue`.

Reply:
Leave for now, will figure it out later